modin/core/dataframe/pandas/partitioning/axis_partition.py

```diff
@@ -360,3 +516,125 @@ def drain(cls, df: pandas.DataFrame, call_queue: list):
         for func, args, kwargs in call_queue:
             df = func(df, *args, **kwargs)
         return df
+
+    def mask(self, row_indices, col_indices):
+        """
+        Create (synchronously) a mask that extracts the indices provided.
+
+        Parameters
+        ----------
+        row_indices : list-like, slice or label
+            The row labels for the rows to extract.
+        col_indices : list-like, slice or label
+            The column labels for the columns to extract.
+
+        Returns
+        -------
+        PandasOnRayDataframeVirtualPartition
+            A new ``PandasOnRayDataframeVirtualPartition`` object,
+            materialized.
+        """
+        return (
+            self.force_materialization()
+            .list_of_block_partitions[0]
+            .mask(row_indices, col_indices)
+        )
+
+    def to_pandas(self):
+        """
+        Convert the data in this partition to a ``pandas.DataFrame``.
+
+        Returns
+        -------
+        pandas DataFrame.
+        """
+        return self.force_materialization().list_of_block_partitions[0].to_pandas()
+
+    def to_numpy(self):
+        """
+        Convert the data in this partition to a ``numpy.array``.
+
+        Returns
+        -------
+        NumPy array.
+        """
+        return self.force_materialization().list_of_block_partitions[0].to_numpy()
+
+    _length_cache = None
+
+    def length(self):
+        """
+        Get the length of this partition.
+
+        Returns
+        -------
+        int
+            The length of the partition.
+        """
+        if self._length_cache is None:
+            if self.axis == 0:
+                self._length_cache = sum(
+                    obj.length() for obj in self.list_of_block_partitions
+                )
+            else:
+                self._length_cache = self.list_of_block_partitions[0].length()
+        return self._length_cache
+
+    _width_cache = None
+
+    def width(self):
+        """
+        Get the width of this partition.
+
+        Returns
+        -------
+        int
+            The width of the partition.
+        """
+        if self._width_cache is None:
+            if self.axis == 1:
+                self._width_cache = sum(
+                    obj.width() for obj in self.list_of_block_partitions
+                )
+            else:
+                self._width_cache = self.list_of_block_partitions[0].width()
+        return self._width_cache
+
+    def wait(self):
+        """Wait completing computations on the object wrapped by the partition."""
+        pass
+
+    def add_to_apply_calls(self, func, *args, length=None, width=None, **kwargs):
+        """
+        Add a function to the call queue.
+
+        Parameters
+        ----------
+        func : callable or ray.ObjectRef
```

Comment:
Not only Ray references.
```suggestion
        func : callable or ray.ObjectRef
```

Reply:
corrected